src/app/components/bookmark-menu/bookmark-menu.component.ts

```diff
@@ -36,31 +38,46 @@ export class BookmarkMenuComponent implements OnInit {
 
   public topBookmarks: Array<BookMark>;
   public isBookmarkOpen = false;
+  public currentCollections = '';
 
   constructor(
     public dialog: MatDialog,
-    private bookmarkService: ArlasBookmarkService
+    private bookmarkService: ArlasBookmarkService,
+    private startupService: ArlasStartupService
   ) { }
 
   public ngOnInit(): void {
     this.icon = this.icon ? this.icon : 'view_list';
     this.nbTopBookmarks = this.nbTopBookmarks ? this.nbTopBookmarks : 3;
+    this.currentCollections = Array.from(this.startupService.collectionsMap.keys()).join(',');
 
     if (this.bookmarkService.dataBase instanceof BookmarkPersistenceDatabase) {
       (this.bookmarkService.dataBase as BookmarkPersistenceDatabase).dataChange
         .subscribe((bookmarks: { total: number, items: BookMark[] }) => {
-          const sortedBookmark = bookmarks.items.sort((a, b) => {
+          const sortedBookmark = bookmarks.items.filter(bk => {
+            if (bk.collections) {
```

Comment:
bk => bk.collections && bk.collections === this.currentCollections should do the same (but do you really need to check if bk.collections is truthy at first?)